src/Microphysics2M.jl

```diff
@@ -228,4 +240,324 @@ function accretion(
     return A * q_liq * q_rai
 end
 
+# Two-moment bulk microphysics autoconversion, accretion, and self-collection
+# rates from Seifert and Beheng 2001
+
+"""
+    autoconversion(param_set, scheme, q_liq, q_rai, ρ, N_liq)
+
+ - `param_set` - abstract set with Earth parameters
+ - `scheme` - type for 2-moment rain autoconversion parameterization
+ - `q_liq` - cloud water specific humidity
+ - `q_rai` - rain water specific humidity
+ - `ρ` - air density
+ - `N_liq` - cloud droplet number density
+
+Returns a named tupple containing q_rai, n_rai, q_liq, n_liq tendencies due to
+collisions between cloud droplets (autoconversion) for `scheme == SB2001Type`
+"""
+function autoconversion(
+    param_set::APS,
+    scheme::CT.SB2001Type,
+    q_liq::FT,
+    q_rai::FT,
+    ρ::FT,
+    N_liq::FT,
+) where {FT <: Real}
+
+    if q_liq < eps(FT)
+        return (;
+            dq_rai_dt = FT(0),
+            dN_rai_dt = FT(0),
+            dq_liq_dt = FT(0),
+            dN_liq_dt = FT(0),
+        )
+    end
+
+    kc::FT = CMP.kc_SB2001(param_set)
+    ν::FT = CMP.ν_SB2001(param_set)
+    xstar::FT = CMP.xstar_SB2001(param_set)
+
+    L_liq = ρ * q_liq
+    x_liq = L_liq / N_liq
+    τ = FT(1) - q_liq / (q_liq + q_rai)
+    ϕ_au = FT(600) * τ^FT(0.68) * (FT(1) - τ^FT(0.68))^FT(3)
+
+    dL_rai_dt =
+        kc / 20 / xstar * (ν + 2) * (ν + 4) / (ν + 1)^2 *
+        L_liq^2 *
+        x_liq^2 *
+        (1 + ϕ_au / (1 - τ)^2)
+    dN_rai_dt = dL_rai_dt / xstar
+    dL_liq_dt = -dL_rai_dt
+    dN_liq_dt = -2 * dN_rai_dt
+
+    return (;
```

Comment:
I like returning named tuples. I think it makes the code easy to read. Not sure about the performance benefit or cost of it?